app/helpers/pagelets_helper.rb

```diff
@@ -67,6 +67,83 @@ def th_content(col)
     sort col[:key], as: col[:label], default: col[:default_sort] || 'ASC'
   end
 
+  def create_table_preferences
+    table_preference = TablePreference.new(:user => User.current, :name => controller_name, :columns => @selected_columns)
+    table_preference.save
+    table_preference
+  end
+
+  def update_table_preferences(table_preference)
+    table_preference.update(:columns => @selected_columns)
+    table_preference.save
+  end
+
+  def defined_columns
+    Pagelets::Manager.pagelets_at('hosts/_list', :hosts_table_column_header).map do |pt|
+      next unless pt.opts[:key]
+      {
+        key: pt.opts[:key],
+        label: pt.opts[:label],
+        profiles: pt.profiles.map { |pr| pr.label },
+        checked: @selected_columns&.include?(pt.opts[:key].to_s),
+      }
+    end.compact
+  end
+
+  def defined_categories
+    defined_columns.map { |col| col[:profiles] }.flatten.uniq
+  end
+
+  def checked_columns(category)
+    all_checked = false
+    category.each do |column|
+      if column[:checkProps][:checked]
+        all_checked = true
+      elsif all_checked
+        all_checked = nil
+        break
+      end
+    end
+    all_checked
+  end
```

Comment:
Since this method returns boolean only, it should be renamed to give more information. Also it can be quite simplified:
```suggestion
  def all_checked?(category)
    !category.any? { |column| !column[:checkProps][:checked] }
  end
```